src/mjolnir/shortcutbuilder.cc

```diff
@@ -366,13 +365,18 @@ uint32_t AddShortcutEdges(GraphReader& reader,
                           const GraphId& start_node,
                           const uint32_t edge_index,
                           const uint32_t edge_count,
-                          std::unordered_map<uint32_t, uint32_t>& shortcuts) {
+                          std::unordered_map<uint32_t, uint32_t>& shortcuts,
+                          std::unordered_set<GraphId>& last_nodes) {
   // Shortcut edges have to start at a node that is not contracted - return if
-  // this node can be contracted.
+  // this node can be contracted and is not a terminus of a pervious shortcut.
   EdgePairs edgepairs;
```

Comment:
I'd leave a TODO here to tell ourselves that we _could_ improve on this by looping over all `last_nodes` after processing all tiles, else it's random whether a broken shortcut will ever be picked up in another tile